pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -2350,7 +2349,7 @@ func (cc *XController) Cleanup(appwrapper *arbv1.AppWrapper) error {
 		if appwrapper.Spec.AggrResources.GenericItems != nil {
 			for _, ar := range appwrapper.Spec.AggrResources.GenericItems {
 				genericResourceName, gvk, err00 := cc.genericresources.Cleanup(appwrapper, &ar)
-				if err00 != nil && !apierrors.IsNotFound(err00) {
+				if err00 != nil && !apierrors.IsNotFound(err00) && !apierrors.IsInvalid(err00) && !IsJsonSyntaxError(err00) {
```

Comment:
Nit: May be combine multiple apierrors as a single error. I wonder if there is cleaner way to check different types of erros, if not may be create a another issue for design of error handling.

Reply:
I am not sure how you'd combine them as they are separate types in the API itself. Are you concerned with the overly long line ? if yes.. then we can probably fix that by using a helper function `DoNotRetryError(err)`. This something you want to pursue ?

Reply:
Yes, a helper function would be great.

Reply:
Done, see the latest commits @asm582 !